docs/HowTo/write/hints.md

```diff
@@ -0,0 +1,54 @@
+---
+description: Using hints for off-circuit computations
+---
+
+# Compiler hints
+
+Instead of computing some value within a circuit, it is sometimes more optimal
+to compute the value off-circuit and only verify the correctness of the
+computation in a circuit. `gnark` provides a way to provide such verifiable
+"advice wires" in the way of *hints*. From the prover point of view, these are
+essentially input variables which are provided by hint function instead of the
+user.
+
+For example, lets consider a decomposition of an integer `a` into bits. A naive
+way to do it would be to look at the binary representation of the integer and
+extract bits from this representation, but `gnark` does not currently provide
+native bit operations. Instead, the hint function `hint.IthBit` can provide the
+bits as variables and the user must constrain these variables as a weighted sum
+which equals to `a`. In a circuit it would look like:
```

Comment:
```suggestion
For example, consider a decomposition of an integer `a` into bits. A naive
way to decompose it is to look at the binary representation of the integer and
extract bits from this representation, but `gnark` doesn't currently provide
native bit operations. Instead, the hint function `hint.IthBit` can provide the
bits as variables and the user must constrain these variables as a weighted sum
which equals to `a`. In a circuit it would look like:
```